rados/rados.go

```diff
@@ -25,6 +25,33 @@ const (
 	RadosAllNamespaces = AllNamespaces
 )
 
+// OpFlags are flags that can be set on a per-op basis.
+type OpFlags uint
```

Comment:
I'm a bit concerned about the naming of this given that in PR #315 we are adding `OperationFlags` and here we will have `OpFlags`. Unfortunately ceph's api isn't much help here as the former are named like LIBRADOS_OPERATION_* and the latter like LIBRADOS_OP_FLAG_*.
The function arguments that take thse are both named `flags` with an `int` type. And yes, they utimately both pertain to rados operations... but for different calls!  
I think we need to brainstorm a bit on this.

Reply:
right, this ugly and gets confusing :disappointed:

Reply:
In librados the flags you are adding here are used in the calls `rados_write_op_set_flags` and `rados_read_op_set_flags`. These calls side-effect the kind of op functions you would have previously used, something like `rados_write_op_writesame`.   The flags already in #315 are used in calls like `rados_write_op_operate` and `rados_read_op_operate`.  So I'm thinking we can maybe keep these names and better distinguish the flags in the other PR with the name `WholeOperationFlags`, as they are passed to the _operate call and thus effect the whole operation. Thoughts?

Reply:
Or maybe just `OpFlags` and `OperateFlags`, indicating the `OperateFlags` are for the `Operate()` call? The `Whole`-prefix looks a little strange. Sticking to the current state is fine with me too.

Reply:
Maybe. "OperateFlags" differs from "Operation" in spelling. (yeah, "Whole" is new but it is a prefix and won't mess up a grep :-) ) 
I suppose "OperationFlags" and "OpFlags" are distinct enough in code... it just looked odd and confusing to me initially. I suppose the status quo could be the least worst option.